website/blog/2024/12-02-scrape-crunchbase/index.md

```diff
@@ -0,0 +1,486 @@
+            'values': ['company']
+        }
+    ]
+}
+
+payload = json.dumps(payload)
+await crawler.run([
+    Request.from_url(
+        url="https://api.crunchbase.com/api/v4/searches/organizations",
+        method="POST",
+        payload=payload,
+        use_extended_unique_key=True,
+        headers=HttpHeaders({'Content-Type': 'application/json'}),
+        label="search"
+    )
+])
+```
+
+For processing search results and pagination, we use the following handler:
+
+```python
+@router.handler('search')
+async def search_handler(context: HttpCrawlingContext) -> None:
+    """Search results handler with pagination support."""
+    context.log.info(f'search_handler processing {context.request} ...')
+
+    data = json.loads(context.http_response.read())
+
+    last_entity = None
+    results = []
+
+    for entity in data['entities']:
+        last_entity = entity['uuid']
+        results.append({
+            "Company Name": entity['properties']["identifier"]["value"],
+            "Short Description": entity['properties']["short_description"],
+            "Website": entity['properties'].get("website_url"),
+            "Location": "; ".join([item["value"] for item in entity['properties'].get("location_identifiers", [])]),
+        })
+
+    if results:
+        await context.push_data(results)
+
+    if last_entity:
+        payload = json.loads(context.request.payload)
+        payload["after_id"] = last_entity
+        payload = json.dumps(payload)
+
+        await context.add_requests([
+            Request.from_url(
+                url="https://api.crunchbase.com/api/v4/searches/organizations",
+                method="POST",
+                payload=payload,
+                use_extended_unique_key=True,
+                headers=HttpHeaders({'Content-Type': 'application/json'}),
+                label="search"
+            )
+        ])
+```
+
+### Free API Limitations
+
+The free version of the API has significant limitations:
+
+- Limited set of available endpoints
+- Autocompletes function only works for company searches
+- Not all data fields are accessible
+- Limited search filtering capabilities
+
+For serious projects, a paid subscription is recommended. Despite these limitations, the API remains the most reliable way to obtain data from Crunchbase.
+
+## Conclusion
+
+We've explored three different approaches to obtaining data from Crunchbase:
+
+1. **Sitemap** - for large-scale data collection
+2. **Search** - difficult to automate due to Cloudflare protection
+3. **Official API** - the most reliable solution for commercial projects
+
+Each method has its advantages, but for most projects, I recommend using the official API despite its limitations in the free version.
+
+The complete source code for all described solutions is available in my [repository](https://github.com/Mantisus/crunchbase-crawlee).
+
+I'd be happy to discuss in the comments which approach seems optimal for your needs and why.
```

Comment:
i loved this blog, i just think it is kind of missing initially what should user expect, as it was not mentioned enough we are going to explore all three ways, and also whats exactly are we doing.

Also if we can breakdown the steps initially it will make the blog more guided.

good work